state/types.go

```diff
@@ -84,7 +85,7 @@ func (s1 *Service) Console() string {
 
 // Equal returns true if s1 and s2 are equal.
 func (s1 *Service) Equal(s2 *Service) bool {
-	return reflect.DeepEqual(s1.Service, s2.Service)
+	return s1.EqualWithOpts(s2, false, true)
```

Comment:
This should be `(s2, false, false)`. The contract for this function is that it compares the two objects without ignoring any fields. This comment also applies to all other `Equal()`.

Reply:
Gotcha--misunderstood and thought we wanted `Equal()` to match our use of `EqualWithOpts()` in practice. Fixed now; forced since the commit message was inaccurate otherwise.